### PR TITLE
Df 512 user info nach personennummer suchen 

### PR DESCRIPTION
**Pull request by dvoracek-martin**

### CODE REVIEW

File: alv-portal-ui/src/app/shared/backend-services/user-info/user-info-repository.ts [hunk 1]
```diff
@@ -25,9 +26,14 @@ export class UserInfoRepository {
     return this.http.post<void>(`${UserInfoRepository.USER_INFO_URL}${userId}/company-contact-templates`, companyContactTemplate);
   }
 
-  public loadUserByEmail(email: string): Observable<UserInfoDTO> {
-    const params = new HttpParams().set('eMail', email);
-    return this.http.get<UserInfoDTO>(UserInfoRepository.USER_INFO_URL, { params: params });
+  public loadUserByEmailOrStesNr(searchparam: string, searchparamType: UserSearchParameterTypes): Observable<UserInfoDTO> {
```
Comment (pavax) [inline alv-portal-ui/src/app/shared/backend-services/user-info/user-info-repository.ts]: repository should be "stupid" so there should not be any logic

File: alv-portal-ui/src/app/admin/user-info/user-info.types.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export enum UserSearchParameterTypes {
+  EMAIL = 'EMAIL',
+  STES_NR= 'STES_NR'
```
Comment (birom) [inline alv-portal-ui/src/app/admin/user-info/user-info.types.ts]: Please check your editor settings. Should be: `` STES_NR = 'STES_NR'``

File: alv-portal-ui/src/app/shared/backend-services/user-info/user-info-repository.ts [hunk 3]
```diff
@@ -25,9 +26,14 @@ export class UserInfoRepository {
...
+    if (searchparamType==UserSearchParameterTypes.EMAIL){
```
Comment (birom) [inline alv-portal-ui/src/app/shared/backend-services/user-info/user-info-repository.ts]: Must be ``searchparamType === UserSearchParameterTypes.EMAIL``

File: alv-portal-ui/src/app/admin/user-info/user-info.component.ts [hunk 4]
```diff
@@ -171,4 +195,22 @@ export class UserInfoComponent extends AbstractSubscriber implements OnInit {
       });
   }
 
+  usedSearchParameterRadioChange() {
+    if (this.form.value.usedSearchParameterRadio===UserSearchParameterTypes.EMAIL){
```
Comment (birom) [inline alv-portal-ui/src/app/admin/user-info/user-info.component.ts]: Please check formatting.

File: alv-portal-ui/src/app/admin/user-info/user-info.component.ts [hunk 5]
```diff
@@ -171,4 +195,22 @@ export class UserInfoComponent extends AbstractSubscriber implements OnInit {
...
+      this.formPlaceholder = 'portal.admin.user-info.use.search.placeholders.email';
+      this.formLabel='portal.admin.user-info.user-info.email';
```
Comment (birom) [inline alv-portal-ui/src/app/admin/user-info/user-info.component.ts]: Please check formatting.

File: alv-portal-ui/src/app/admin/user-info/user-info.component.ts [hunk 6]
```diff
@@ -70,17 +71,34 @@ export class UserInfoComponent extends AbstractSubscriber implements OnInit {
...
+  ]);
 
   ngOnInit() {
     this.form = this.fb.group({
-      emailAddress: [null, [Validators.required, patternInputValidator(EMAIL_REGEX)]]
+      searchParam: [null, [Validators.required, patternInputValidator(EMAIL_REGEX)]],
+      usedSearchParameterRadio: [UserSearchParameterTypes.EMAIL]
     });
+    this.formPlaceholder = 'portal.admin.user-info.use.search.placeholders.email';
+    this.formLabel='portal.admin.user-info.user-info.email';
```
Comment (birom) [inline alv-portal-ui/src/app/admin/user-info/user-info.component.ts]: Please check formatting.

File: alv-portal-ui/src/app/admin/user-info/user-info.component.ts [hunk 7]
```diff
@@ -171,4 +200,27 @@ export class UserInfoComponent extends AbstractSubscriber implements OnInit {
       });
   }
 
+  private searchParameterTypeChanged(newValue: UserSearchParameterTypes) {
+    let validator = null;
```
Comment (pavax) [inline alv-portal-ui/src/app/admin/user-info/user-info.component.ts]: [suggestion] let's rename that variable to searchParamValidator

File: alv-portal-ui/src/app/admin/user-info/user-info.component.ts [hunk 8]
```diff
@@ -70,16 +75,39 @@ export class UserInfoComponent extends AbstractSubscriber implements OnInit {
...
   ngOnInit() {
     this.form = this.fb.group({
-      emailAddress: [null, [Validators.required, patternInputValidator(EMAIL_REGEX)]]
+      searchParam: [null],
+      searchParameterType: [UserSearchParameterTypes.EMAIL]
+    });
+
+    this.form.get('searchParameterType').valueChanges.pipe(
+      startWith(this.form.get('searchParameterType').value),
+      distinctUntilChanged()
```
Comment (pavax) [inline alv-portal-ui/src/app/admin/user-info/user-info.component.ts]: we need to unsubscribe once the component is destroyed -> see AbstractSubscriber

File: alv-portal-ui/translations.csv [hunk 9]
```diff
@@ -701,6 +701,8 @@ portal.admin.user-info.desc,Bitte geben Sie die E-Mail-Adresse eines bestehenden
 portal.admin.user-info.eIAM-roles.no,Es konnte keine Rollen in eIAM für diesen Benutzer gefunden werden,No roles for this user found in eIAM,No roles for this user found in eIAM,No roles for this user found in eIAM,
 portal.admin.user-info.eIAM-roles.title,eIAM Rollen,eIAM Roles,eIAM Roles,eIAM Roles,
 portal.admin.user-info.email,E-Mail,Email,Courriel,E-mail,
+portal.admin.user-info.email,E-Mail,Email,Courriel,E-mail,
+portal.admin.user-info.emailOrPersonalNumber,E-Mail/Personennummer,Email/Personal ID,Courriel/Numéro personnel,E-mail/Numero personale,
```
Comment (pavax) [inline alv-portal-ui/translations.csv]: for sake of completeness please add the new keys to the jira issue: DF-598

File: alv-portal-ui/src/app/admin/user-info/user-info.component.ts [hunk 10]
```diff
@@ -1,18 +1,23 @@
...
-import { UserInfoDTO } from '../../shared/backend-services/user-info/user-info.types';
-import { HttpErrorResponse } from '@angular/common/http';
-import { EMPTY } from 'rxjs';
-import { catchError, switchMap, takeUntil } from 'rxjs/operators';
-import { UserRole } from '../../core/auth/user.model';
-import { Notification, NotificationType } from '../../shared/layout/notifications/notification.model';
-import { UserInfoBadge, UserInfoBadgesMapperService } from './user-info-badges-mapper.service';
-import { ModalService } from '../../shared/layout/modal/modal.service';
-import { ConfirmModalConfig } from '../../shared/layout/modal/confirm-modal/confirm-modal-config.model';
+import {Component, OnInit} from '@angular/core';
```
Comment (birom) [inline alv-portal-ui/src/app/admin/user-info/user-info.component.ts]: Please update your editor setting as discussed. Thx

**Pull request merged**